src/vm/amd64/jithelpers_fastwritebarriers.S

```diff
@@ -244,3 +246,331 @@ PATCH_LABEL JIT_WriteBarrier_SVR64_PatchLabel_CardTable
         mov     byte ptr [rdi + rax], 0FFh
         ret
 LEAF_END_MARKED JIT_WriteBarrier_SVR64, _TEXT
+
+#endif
+
+
+#ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
+
```

Comment:
I'm not sure in what scenarios the different write barriers helped originally when they were written. I assumed there are scenarios where each is useful. The benchmarks I'm running spend relatively little time in write barriers, and I don't see any noticeable perf difference between PreGrow64 and PostGrow64. I don't see the 32-bit constant write barriers being used at all normally, but not sure if there's a scenario where they would be used, and how much better perf they would give.